sdk/documenttranslator/ai-document-translator-rest/review/ai-document-translator.api.md

```diff
@@ -0,0 +1,745 @@
+    // (undocumented)
+    body: DocumentStatusResponse;
+    // (undocumented)
+    headers: RawHttpHeaders & GetOperationDocumentsStatus200Headers;
+    // (undocumented)
+    status: "200";
+}
+
+// @public
+export interface GetOperationDocumentsStatus400Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "400";
+}
+
+// @public
+export interface GetOperationDocumentsStatus401Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "401";
+}
+
+// @public
+export interface GetOperationDocumentsStatus404Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "404";
+}
+
+// @public
+export interface GetOperationDocumentsStatus429Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "429";
+}
+
+// @public
+export interface GetOperationDocumentsStatus500Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "500";
+}
+
+// @public
+export interface GetOperationDocumentsStatus503Response extends HttpResponse {
+    // (undocumented)
+    body: ErrorResponseV2;
+    // (undocumented)
+    status: "503";
+}
+
+// @public (undocumented)
+export type GetOperationDocumentsStatusParameters = RequestParameters & GetOperationDocumentsStatusQueryParam;
+
+// @public (undocumented)
+export interface GetOperationDocumentsStatusQueryParam {
+    // (undocumented)
+    queryParameters?: GetOperationDocumentsStatusQueryParamProperties;
+}
+
+// @public (undocumented)
+export interface GetOperationDocumentsStatusQueryParamProperties {
+    $maxpagesize?: number;
+    $orderBy?: string[];
+    $skip?: number;
+    $top?: number;
+    createdDateTimeUtcEnd?: Date;
+    createdDateTimeUtcStart?: Date;
+    ids?: string[];
+    statuses?: string[];
+}
+
+// @public (undocumented)
+export interface GetOperations {
+    get(options?: GetOperationsParameters): Promise<GetOperations200Response | GetOperations400Response | GetOperations401Response | GetOperations429Response | GetOperations500Response | GetOperations503Response>;
+    post(options?: SubmitBatchRequestParameters): Promise<SubmitBatchRequest202Response | SubmitBatchRequest400Response | SubmitBatchRequest401Response | SubmitBatchRequest429Response | SubmitBatchRequest500Response | SubmitBatchRequest503Response>;
+}
+
+// @public (undocumented)
+export interface GetOperations200Headers {
+    "retry-after"?: string;
+    etag?: string;
+}
+
+// @public
+export interface GetOperations200Response extends HttpResponse {
+    // (undocumented)
+    body: BatchStatusResponse;
+    // (undocumented)
+    headers: RawHttpHeaders & GetOperations200Headers;
+    // (undocumented)
+    status: "200";
+}
+
+// @public
+export interface GetOperations400Response extends HttpResponse {
```

Comment:
Seems like the strong type of `status` isn't doing much here. We end up with a lot of interfaces that contain the exact same information that just have names for which status code they are returning. Seems like an enum use case if what we want is to strictly specify the set of possible `status` values for errors.

Reply:
Yeah, this is a great point. I'll try to collapse these identical interface into a single one. I'll have to get this working in the generator first. I may not be able to fit it in this PR but I'll definitely address this when sending the Generator PR